pandas/_libs/algos_take_helper.pxi.in

```diff
@@ -184,7 +107,29 @@ def take_1d_{{name}}_{{dest}}(ndarray[{{c_type_in}}, ndim=1] values,
     # We cannot use the memoryview version on readonly-buffers due to
```

Comment:
Is this still valid? Seems like making these functions `cpdef` would solve some of the problems I guess this was trying to solve originally

Reply:
looking the blame, i think you're right that this is outdated, but not that cpdef would help.  i think using the `const` modifier would handle this in most of the relevant cases, but we'll still need to do some gymnastics with `object[:]`